### DISCUSSION TITLE
GraphQL compatibility in native-images

### DISCUSSION THREAD

**Discussion by eray-kaya** [graphql-java-kickstart/graphql-spring-boot#560] [Q&A]
Hi,

i am currently trying to create a working native-image for a graphql-spring-project using graalvm.
Creating the native-image is no problem. However, trying to run the native-image an error occurs
which is caused by the AutoConfiguration classes of graphql and graphiql. The error can be found at the bottom. As i know, its mandatory to set proxyBeanMethods = false in spring-boot-applications to create successfully native-images.
Therefore, i am not able to create a working native-image due to the fact that graphql uses proxyBeanMethods which are not set false when the @SpringBootApplication annotation is set false for proxyBeanMethods. 
I wasn't able find a thread that contained the compatibility of graphql and native-images
which is why I would like to ask, if my assumptions are right and if thats the case, is there currently a roadmap
to provide compatibility using graphql in native-images?

Error-line when the native-image is executed:
org.springframework.beans.factory.BeanDefinitionStoreException: @Configuration classes need to be marked as proxyBeanMethods=false. Found: [graphql.kickstart.graphiql.boot.GraphiQLAutoConfiguration, graphql.kickstart.tools.boot.GraphQLJavaToolsAutoConfiguration, graphql.kickstart.spring.web.boot.GraphQLInstrumentationAutoConfiguration]

graphql dependencies that are used in the project:

implementation 'com.graphql-java-kickstart:graphiql-spring-boot-starter:7.2.0'
implementation 'com.graphql-java-kickstart:graphql-spring-boot-starter:7.2.0'

**Comment by miguelangellv** [comment]
Did you get any solution?
